base/strings/util.jl

```diff
@@ -248,6 +248,123 @@ julia> rpad("March",20)
 rpad(s, n::Integer, p=" ") = rpad(string(s),n,string(p))
 cpad(s, n::Integer, p=" ") = rpad(lpad(s,div(n+strwidth(s),2),p),n,p)
 
+immutable SplitIterator
+    str::AbstractString
+    splitter
```

Comment:
is there any algorithmic reason why `eachsplit` is slower than `split`? would parameterizing the `SplitIterator` on the types of `str` and `splitter` remove that gap, and allow to define `split(...) = collect(eachsplit(...))`?